### DISCUSSION TITLE
Multi language forms

### DISCUSSION THREAD

**Discussion by fuerst** [Budibase/budibase#5834] [Help]
Dear all, do you got any idea how to show a form at different languages? I want to address users at more than one country. Thanks a lot!

**Comment by melohagan** [comment]
Hi @fuerst 

This is a good idea, and it has been raised in a couple of issues:

- https://github.com/Budibase/budibase/issues/3950
- https://github.com/Budibase/budibase/issues/3281

The manual way of doing this would be to use Conditional UI: https://docs.budibase.com/docs/conditions to **Show** a container that contains a form with the language of choice. 

You could have a dropdown on your home page that allows the user to select language and updates and *persists* the state: https://docs.budibase.com/docs/app-state

**Reply by fuerst** [reply]
Thanks, looking forward for those issues to be solved.

**Comment by dannyshin76** [comment]
My implementation, to avoid maintaining multiple forms for each language:

1. In the layout, add language selection UI. Use State variable to hold the language selection.
2. Create a multi-language STRING table in the Budibase DB, with columns of string keys and each language that you want to support.
3. Include STRING data provider at the root of every screen.
4. In screens and forms, use handlebar to display appropriate DB strings of selected language wherever you want (such as labels and info texts).

Handlebar expression example:

`{{#forEach STRINGS.Rows }}{{#eq KEY 'LABEL_YEAR'}}{{#eq ../State.language 'KO'}}{{KO}}{{else}}{{EN}}{{/eq}}{{/eq}}{{/forEach}}`

As you can naturally infer, the STRING table has 3 columns: KEY, KO (for Korean), and EN (for English). This expression will display "연도" or "Year" based on user's language selection. If the user didn't choose a language, EN will be displayed as a default.

This approach pretty much worked for me. Hope this will help you :)

**Reply by melohagan** [reply]
Awesome! Good idea :)

**Reply by fuerst** [reply]
Thanks for the suggestion using a String table. That sounds promising, will give it a try.

**Reply by marian001** [reply]
Great idea! 
How did you deal with navigation bar? It seems navigation bar link labels don't accept bindings/handlebars...
